index.js

```diff
@@ -167,6 +211,17 @@ const line = []
 // console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 3`
 
+function takeATicketNumber(arr) {
+  if (arr.length == 0) {
+    arr.push(1);
+    console.log("Welcome. You are ticket number 1")
+  } else {
+  let next = arr[arr.length - 1] + 1 
+  arr.push(next);
+  console.log(`Welcome. You are ticket number ${next}`);
+  }
+}
```

Comment:
For this function, we'd need another variable outside the scope of the function to get the desired functionality:

```js
let ticketNumber = 0

function takeATicketNumber(line) {
  ticketNumber++
  line.push(ticketNumber)
  return `Welcome. You are ticket number ${ticketNumber}`
}
```